R/shinywrappers.R

```diff
@@ -295,6 +333,20 @@ renderImage <- function(expr, env=parent.frame(), quoted=FALSE,
     imageOutput, outputArgs)
 }
 
+isTempFile <- function(path, tempDir = tempdir(), default = FALSE) {
+  if (nchar(tempDir) == 0) {
+    # This should never happen, but just to be super paranoid...
+    return(default)
+  }
+  tempDir <- normalizePath(tempDir, winslash = "/", mustWork = FALSE)
+  if (substr(tempDir, nchar(tempDir), nchar(tempDir)) != .Platform$file.sep) {
```

Comment:
Not super important, but there's a function `ensure_trailing_slash` that does the same thing and conveys intent better. (Also `.Platform$file.sep` is apparently `/` on all platforms.)